kubernetes/kubernetes.go

```diff
@@ -326,6 +352,34 @@ func (in *K8SClient) GetPods(namespace, labelSelector string) ([]core_v1.Pod, er
 	}
 }
 
+// GetPodPortForwarder returns a port-forwarder struct which represents an open server forwarding request to the
+// requested pod and port
+// namespace: name of the namespace where the pod lives in.
+// name: name of the pod living in the namespace
+// portMap: ports open by the forwarder. Local port and destination port. Format: "80:8080" (local:destination)
+// It returns both a portforwarder and an error (if present)
+func (in *K8SClient) GetPodPortForwarder(namespace, name, portMap string) (*httputil.PortForwarder, error) {
+	writer := new(bytes.Buffer)
+
+	clientConfig, err := ConfigClient()
+	if err != nil {
+		log.Errorf("Error getting Kubernetes Client config: %v", err)
+		return nil, err
+	}
+
+	// First try whether the pod exist or not
+	_, err = in.GetPod(namespace, name)
```

Comment:
Perhaps worth here to fail on the pod phase [1]

![image](https://user-images.githubusercontent.com/1662329/124885767-e485ce80-dfd3-11eb-89f7-e46e75962fb4.png)

That may save some cycles on initializations and in case something is odd, to not propagate the timeout of the http.

[1] https://github.com/kubernetes/kubernetes/blob/v1.6.0-alpha.0/pkg/kubectl/cmd/portforward.go#L107